### ISSUE TITLE
Allow scroll direction setting for mice and trackpad separately

### ISSUE THREAD

**Issue by mikeweilgart** [folivoraAI/BetterTouchTool#2366]
#### Description of feature request:

Apple's so-called "natural" scroll direction is in fact pretty natural for trackpads.  It is highly counter-intuitive for an actual mouse with a scroll wheel.

Unfortunately, in the Apple system preferences, there is only one scroll direction setting.  (Actually it appears under both the "Mouse" preference pane and the "Trackpad" preference pane, but if you set one of them to "natural" the other one changes as well.)

It would be wonderful if the scroll direction could be set separately for the mouse and the trackpad.

I played around with the "normal mice" triggers, but although it can detect "scroll down," I can't find a way to map that to "scroll up."

----

For what it's worth, the Apple stack exchange site question on this exact point has had 2600+ views in 5 years.  https://apple.stackexchange.com/q/70868/151730

In actual fact there is nothing wrong with the solution presented there, I would just love it if this were integrated into BTT.  And the scroll-reverser software mentioned is under the Apache license, so it may even be possible (and legal) for you to just integrate their code.  :)  https://github.com/pilotmoon/Scroll-Reverser

**Comment by fifafu** [comment]
In the advanced settings is a checkbox to only apply the BTT reverse scrolling to normal mice. I'm not at home currently, if you can't find the option I'll post a screenshot this evening.

**Comment by mikeweilgart** [comment]
Aha, you're right!  Thanks!  However....

I've just played with it and it has the additional effect of turning off the variable sensitivity of scrolling; in other words, increasing the granularity.  So that each individual tiny motion of the scroll wheel (each increment) causes the screen to move the same (large) amount, rather than causing a very tiny motion of window if scrolled very slowly, and causing more motion of window (per increment) if you move the wheel quickly.  (This is much easier to recognize than to describe.)

Would it be possible to adjust the code so that it doesn't have this side effect on normal mice?  (The Scroll-Reverser software doesn't have this effect, so it should be possible; I've no idea how the effect is achieved at the code level, though.)

-----

Experiment that can demonstrate what I'm talking about (steps to reproduce):

1. Turn off the checkbox in advanced settings if it's on.  (Or just turn off BTT.)
2. Go to a website that's very long (but not infinite scroll).
3. Place your finger at the front end of your scroll wheel (next to the cord, if you have one).
4. *Quickly* move your finger to the back end of your scroll wheel (scroll that exact distance).
5. Note how far you have moved on the page.
6. Return to the top of the page.
7. Make the same exact amount of scroll wheel motion (move your finger the same distance on the wheel), but do it very slowly.  Note the difference in how far you moved down the page.
8. Turn on the BTT checkbox and repeat these steps.  You will see that with the checkbox on, you move the same distance down the page regardless of how fast you move the wheel.

In other words, you don't have the same fine-grained control and automatic fast scrolling.

Is that something you could look into adjusting?

**Comment by yisyang** [comment]
@mikeweilgart with regard to the original ticket here, you can find the solution in previous ticket:
https://github.com/fifafu/BetterTouchTool/issues/1240

There was a bug in version 2.424 (781) that made the option not work, but 2.425 (782) fixed it.
